subscriptions-matcher/src/main/java/io/arlas/subscriptions/service/KafkaConsumerRunner.java

```diff
@@ -58,34 +59,56 @@
+                                logger.debug("Received subscription event:" + event.toString());
 
-                    try {
-                        final SubscriptionEvent event = objectMapper.readValue(record.value(), SubscriptionEvent.class);
-                        logger.debug("Received subscription event:" + event.toString());
+                                long t0 = System.currentTimeMillis();
+                                List<Hit> hits = subscriptionsService.searchMatchingSubscriptions(event);
+                                duration = System.currentTimeMillis() - t0;
+                                logger.debug("Nb subscription matching (took " + duration + "ms)=" + hits.toString());
 
-                        List<Hit> hits = subscriptionsService.searchMatchingSubscriptions(event);
-                        logger.debug("Subscription matcher result=" + hits.toString());
+                                t0 = System.currentTimeMillis();
+                                productService.processMatchingProducts(event, hits);
+                                duration = System.currentTimeMillis() - t0;
+                                logger.debug("Product matching result (in " + duration + "ms)");
 
-                        productService.processMatchingProducts(event, hits);
-
-                    } catch (IOException|ParseException e) {
-                        logger.warn("Could not parse record " + record.value(), e);
+                            } catch (IOException | ParseException e) {
+                                logger.warn("Could not parse record " + record.value(), e);
+                            }
+                        }
+                        consumer.commitSync();
+                        nbFailure = 0;
+                    }
+                } catch(CommitFailedException e) {
+                    nbFailure++;
+                    duration = System.currentTimeMillis() - start;
+                    logger.warn("[" + kafkaConfiguration.subscriptionEventsTopic + "] Commit failed (attempt nb " + nbFailure + "): process time=" + duration + "ms (compare to max.poll.interval.ms value) / exception=" + e.getMessage());
+                    if (nbFailure > kafkaConfiguration.commitMaxRetries) {
+                        logger.error("[" + kafkaConfiguration.subscriptionEventsTopic + "] Too many attempts, exiting.");
+                        consumer.close();
+                        System.exit(1);
                     }
                 }
-                consumer.commitSync();
             }
-
         } catch (WakeupException e) {
             // Ignore exception if closing
             if (!closed.get()) throw e;
         } finally {
-            logger.error("Closing consumer of topic '" + kafkaConfiguration.subscriptionEventsTopic + "'");
+            logger.error("["+kafkaConfiguration.subscriptionEventsTopic+"] Closing consumer");
             consumer.close();
```

Comment:
May the .close() throw in exception (in this case, application would not exit?)

Reply:
Ok, catching Runtime to avoid any surprise